### ISSUE TITLE
Search in selection does not support multi-selections

### ISSUE THREAD

**Issue by Simran-B** [microsoft/vscode#38363]
- VSCode Version: 1.18.0
- OS Version: Windows 10

Steps to Reproduce:

1. Search for a phrase (Ctrl+F) that matches multiple times, e.g. `FOO` in example text:

       FOO bar baz
       one two three FOO four
       123 FOO 456 789

    ![search](https://user-images.githubusercontent.com/7819991/32812560-f46ade00-c9a5-11e7-882c-be36e9ca553c.png)

2. Selection menu > Select all occurrences (Ctrl+Shift+L)

3. Enable Search in selection ![image](https://user-images.githubusercontent.com/7819991/32812781-19432556-c9a7-11e7-9852-2cfcc77f12a4.png) in the search panel (Alt+L)

4. The multi-selection is removed, and only the active line will be searched from there on

Expected behavior:

Search/replace should be constrained to the three selections of the word `FOO` instead of just one if Select all occurrences was used (so it's an actual multi-selection, as opposed to the match highlighting of the search).

Use case:

In my case, I tried to use a regex to create a complex selection, then restrict a regex-based replacement action to that selection. More specifically, turn something like
`{ "foo": 1, "bar": [ 2, 3, 4, 5 ] }` into
`{ "foo": 1, "bar": [2,3,4,5] }`. The selection regex was supposed to restrict the replacement of spaces with nothing to the array part `[ ... ]`.

Reproduces without extensions: Yes

**Comment by Simran-B** [comment]
I guess the expected behavior would be, that search & replace is executed individually for every selection. Pseudo-code:

```
For Selection of Selections:
    Selection.Content = Selection.Content.replace(Find.search, Find.replace)
```

Or should adjacent selections be treated as one (corner case)?

![image](https://user-images.githubusercontent.com/7819991/32943447-704fad42-cb8c-11e7-8503-da299eff133e.png)

Overlapping selections are not possible to my knowledge.

**Comment by rebornix** [comment]
This feature request will not be considered in the next [6-12 months roadmap](https://github.com/Microsoft/vscode/wiki/Roadmap) and as such will be closed to keep the number of issues we have to maintain actionable. Thanks for understanding and happy coding!

**Comment by assumptionsoup** [comment]
To be honest, I thought this was a bug not a feature when I encountered it. Nothing in the editor makes it clear why this doesn't work, and it's incredibly frustrating trying to get the editor to do this.  Search-in-selection seems to disable your selection (or at least makes it very unclear where your selection went).  The fact that I had to go back to sublime text to accomplish this task this should speak volumes.
